src/robot/libraries/Collections.py

```diff
@@ -444,14 +470,23 @@ def lists_should_be_equal(
             values,
         )
 
-    def _get_list_index_name_mapping(self, names, list_length):
+    def _get_list_index_name_mapping(
+        self,
+        names: "Mapping[int, str] | Iterable[str] | None",
+        list_length: int,
+    ) -> "dict[int, str]":
         if not names:
             return {}
         if is_dict_like(names):
-            return {int(index): names[index] for index in names}
-        return dict(zip(range(list_length), names))
+            return {int(index): names[index] for index in names}  # type: ignore
+        return dict(zip(range(list_length), names))  # type: ignore
 
-    def _yield_list_diffs(self, list1, list2, names):
+    def _yield_list_diffs(
+        self,
+        list1: Iterable,
+        list2: Iterable,
+        names: "dict[int, str]",
+    ) -> Generator[str, None, None]:
```

Comment:
- The code always passes here sequences, so typing should use `Sequence` instead of `Iterable`.
- `Generator[str, None, None]` is technically correct, but `Iterator[str]` would be simpler and we've used in elsewhere with `yield`. See the docs here for details:
https://docs.python.org/3/library/typing.html#annotating-generators-and-coroutines